src/metabase/automagic_dashboards/filters.clj

```diff
@@ -0,0 +1,94 @@
+(ns metabase.automagic-dashboards.filters
+  (:require [clojure.string :as str]
+            [metabase.models
+             [card :refer [Card]]
+             [dashboard :as dashboard :refer [Dashboard]]
+             [dashboard-card :refer [DashboardCard]]
+             [field :refer [Field]]]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(def ^:private FieldIdForm
+  [(s/one (s/constrained (s/either s/Str s/Keyword)
+                         (comp #{"field-id" "fk->"} str/lower-case name))
+          "head")
+   s/Any])
+
+(defn- collect-field-references
+  [card]
+  (->> card
+       :dataset_query
+       :query
+       ((juxt :breakout :fields))
+       (tree-seq sequential? identity)
+       (remove (s/checker FieldIdForm))))
+
+(defn- candidates-for-filtering
+  [cards]
+  (->> cards
+       (mapcat collect-field-references)
```

Comment:
Does this mean that only fields that are currently in a breakout in a card are eligible for automated filters?

Reply:
Yes. Seemed like a good heuristic to prevent having overbloat and in line with what I expect as a user. If this were implemented MB-wide it would be cool to maybe have a CTA at the end of automagic filters to the effect "add some more of your own filters".